src/content/app/entity-viewer/EntityViewerPage.tsx

```diff
@@ -83,21 +84,30 @@ export const serverFetch: ServerFetch = async (params) => {
   const { data: genomeInfoData, error: genomeInfoError } =
     await genomeInfoResponsePromise;
 
-  // FIXME: 404 status code in response
   if (isGenomeNotFoundError(genomeInfoError)) {
     return {
       status: 404
     };
   }
 
-  const genomeId = genomeInfoData?.genomeId;
+  const genomeId = genomeInfoData?.genomeId as string; // by this point, genomeId clearly exists
 
-  if (!genomeId) {
-    return; // this shouldn't happen
+  // If the url is /entity-viewer/:genomeId, there is nothing more to do
+  if (!entityId) {
+    return;
   }
 
-  // FIXME: gene symbol parsing can explode!
-  const geneStableId = parseFocusIdFromUrl(entityId).objectId;
+  // NOTE: we will have to be smarter here when entities are no longer just genes
+  let geneStableId;
```

Comment:
Can't this be named as `objectStableId` or just `stableId` instead?

Reply:
It won't help much — down below, we are passing this id to the `fetchGenePageMeta.initiate` function, which fetches a gene from Thoas. When we get more entities than just gene, we will have to be able to choose which function to call for different entity types.